add support for bzip2 compression to man2html and friends

--- man2html/glimpse_filters
+++ man2html/glimpse_filters
@@ -1,3 +1,4 @@
+*.bz2	bzip2 -d -c
 *.gz	gzip -d -c
 *.Z	gzip -d -c
 
--- man2html/scripts/cgi-bin/man/man2html
+++ man2html/scripts/cgi-bin/man/man2html
@@ -87,6 +87,9 @@
 if [ -r "$PAGE" ]
 then
     case "$PAGE" in
+    *.bz2)
+	bzcat "$PAGE" | "$MAN2HTML" "$LL" -D "$PAGE"
+	;;
     *.gz)
 	zcat "$PAGE" | "$MAN2HTML" "$LL" -D "$PAGE"
 	;;
@@ -94,6 +97,9 @@
 	"$MAN2HTML" "$LL" "$PAGE"
 	;;
     esac
+elif [ -r "$PAGE".bz2 ]
+then
+	bzcat "$PAGE".bz2 | "$MAN2HTML" "$LL" -D "$PAGE"
 elif [ -r "$PAGE".gz ]
 then
 	zcat "$PAGE".gz | "$MAN2HTML" "$LL" -D "$PAGE"
--- man2html/scripts/cgi-bin/man/mansearch
+++ man2html/scripts/cgi-bin/man/mansearch
@@ -153,7 +153,7 @@
       }
       print "<DT> <a href=\"" cgipath "/man2html?" fullname "\">";
       textname = filename;
-      sub(/\.(gz)|Z|z$/, "", textname);
+      sub(/\.(gz)|(bz2)|Z|z$/, "", textname);
       sub(/\./, "(", textname);
       textname = textname ")";
       print textname;
--- man2html/scripts/cgi-bin/man/mansec
+++ man2html/scripts/cgi-bin/man/mansec
@@ -128,7 +128,7 @@
 				# Print out alphabetic quick index and other links
       }
 				# Split page.n into "page" and "n" and generate an entry
-      sub(/[.]([zZ]|(gz))$/, "", manpage);
+      sub(/[.]([zZ]|(gz)|(bz2))$/, "", manpage);
       match(manpage, /[.][^.]+$/);      
       title = substr(manpage, 1, RSTART - 1);
       if (section != "all") {
